allensdk/brain_observatory/behavior/write_nwb/__main__.py

```diff
@@ -38,13 +37,24 @@ def write_behavior_ophys_nwb(session_data, nwb_filepath):
                      "with a BehaviorOphysSession created from LIMS")
         assert sessions_are_equal(json_session, lims_session, reraise=True)
 
-        BehaviorOphysNwbApi(nwb_filepath_inprogress).save(json_session)
+        BehaviorOphysNwbApi(nwb_filepath_inprogress).save(json_session,
+                                                          skip_eye_tracking=skip_eye_tracking)
 
         logging.info("Comparing a BehaviorOphysSession created from JSON "
                      "with a BehaviorOphysSession created from NWB")
         nwb_api = BehaviorOphysNwbApi(nwb_filepath_inprogress)
         nwb_session = BehaviorOphysSession(api=nwb_api)
-        assert sessions_are_equal(json_session, nwb_session, reraise=True)
+
+        skip_field = set()
+        if skip_eye_tracking:
+            skip_field.add('_eye_tracking')
```

Comment:
I wonder if instead of modifying `sessions_are_equal()` and having the concept of a `skip_field` if you instead just do:

```
if skip_eye_tracking:
    json_session.eye_tracking = None
```

You could also modify the check for line 157 of `BehaviorOphysNwbApi` to only call `add_eye_tracking_data_to_nwb` if `session.eye_tracking is not None`.

Reply:
I don't know why, but this solution does not work. When I include
```
json_session.eye_tracking = None
assert json_session.eye_tracking is None
```
in the code, the assertion fails. I assume there is a setter somewhere that needs to be overridden.

For the sake of speed, I'm just going to fix my current solution (i.e. initialize `skip_field` to `None`). We can revisit this in a later ticket. We will probably want to revisit it, in case we need to rehabilitate data that had other streams anomalously fail.

Reply:
Hmm... you could also try `json_session._eye_tracking = None` since that's what the setter should be accessing.